crates/subspace-networking/src/node_runner.rs

```diff
@@ -289,6 +289,8 @@ where
 
                 self.dial_peer(peer_id, addr)
             }
+        } else {
+            self.networking_parameters_registry.reset_batcher()
```

Comment:
`current_connection_number` above includes pending connections, which are in no way guaranteed to result in successful connection. Thus we need to compare to actually established connections rather than just taking `else` branch here.